datafusion/optimizer/src/analyzer/type_coercion.rs

```diff
@@ -912,15 +912,17 @@ mod test {
     fn agg_udaf_invalid_input() -> Result<()> {
         let empty = empty();
         let return_type = DataType::Float64;
-        let state_type = vec![DataType::UInt64, DataType::Float64];
         let accumulator: AccumulatorFactoryFunction =
             Arc::new(|_| Ok(Box::<AvgAccumulator>::default()));
         let my_avg = AggregateUDF::from(SimpleAggregateUDF::new_with_signature(
             "MY_AVG",
             Signature::uniform(1, vec![DataType::Float64], Volatility::Immutable),
             return_type,
             accumulator,
-            state_type,
+            vec![
+                Field::new("count", DataType::UInt64, true),
```

Comment:
I actually think this is a nice change that the fields are now named rather than anonymous

I think @jacksonrnewhouse mentioned this might be helpful for his project as well